test/e2e/daemon_set.go

```diff
@@ -469,3 +592,64 @@ func checkDaemonPodsImage(c clientset.Interface, ns string, selector map[string]
 		return true, nil
 	}
 }
+
+func templateGenerationMatch(c clientset.Interface, ns string, selector map[string]string, templateGeneration string) func() (bool, error) {
+	return func() (bool, error) {
+		err := checkDaemonPodsTemplateGeneration(c, ns, selector, templateGeneration)
+		match := err == nil
+		return match, nil
```

Comment:
This is the condition func for wait.Poll. If we return an `err` that's not `nil`, we'll immediately exit the wait loop.